core/src/parcel.rs

```diff
@@ -404,12 +404,16 @@ pub struct AssetOutPoint {
     pub amount: u64,
 }
 
+pub const UNLOCK_SUCCESS: u8 = 0x01;
+pub const UNLOCK_BURN: u8 = 0x02;
+
 #[derive(Debug, Clone, Eq, PartialEq, RlpDecodable, RlpEncodable, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct AssetTransferInput {
     pub prev_out: AssetOutPoint,
     pub lock_script: Bytes,
     pub unlock_script: Bytes,
+    pub unlock_type: u8,
```

Comment:
I don't agree with that making this flag.
The most use cases will be just the transfer not burning, and this flag seems verbose.
How about splitting transaction input list?

Reply:
Agreed. I'll fix this.